src/analysis/special_functions/stirling.lean

```diff
@@ -203,3 +209,88 @@ begin
   rw ←filter.tendsto_add_at_top_iff_nat 1,
   exact tendsto_at_top_cinfi stirling_seq'_antitone ⟨x, hx'⟩,
 end
+
+/-!
+ ### Part 2
+ https://proofwiki.org/wiki/Stirling%27s_Formula#Part_2
+-/
+
+/-- For `n : ℕ`, define `w n` as `2^(4*n) * n!^4 / ((2*n)!^2 * (2*n + 1))` -/
+noncomputable def w (n : ℕ) : ℝ :=
+(2 ^ (4 * n) * n.factorial ^ 4) / ((2 * n).factorial ^ 2 * (2 * n + 1))
+
+/-- The sequence `w n` converges to `π/2` -/
+lemma wallis_consequence : tendsto (λ (n : ℕ), w n) at_top (𝓝 (π/2)) :=
+begin
+  convert tendsto_prod_pi_div_two,
+  funext n,
+  induction n with n ih,
+  { rw [w, prod_range_zero, cast_zero, mul_zero, pow_zero, one_mul, mul_zero, factorial_zero,
+        cast_one, one_pow, one_pow, one_mul, mul_zero, zero_add, div_one] },
+  rw [w, prod_range_succ, ←ih, w, _root_.div_mul_div_comm, _root_.div_mul_div_comm],
+  refine (div_eq_div_iff (ne_of_gt _) (ne_of_gt _)).mpr _,
+  { exact mul_pos (pow_pos (cast_pos.mpr (2 * n.succ).factorial_pos) 2)
+      (add_pos (mul_pos two_pos (cast_pos.mpr n.succ_pos)) one_pos) },
+  { let h : 0 < 2 * (n : ℝ) + 1 :=
+    add_pos_of_nonneg_of_pos (mul_nonneg zero_le_two n.cast_nonneg) one_pos,
+    exact mul_pos (mul_pos (pow_pos (cast_pos.mpr (2 * n).factorial_pos) 2) h)
+      (mul_pos h (add_pos_of_nonneg_of_pos (mul_nonneg zero_le_two n.cast_nonneg) three_pos)) },
+  { simp_rw [nat.mul_succ, factorial_succ, succ_eq_add_one, pow_succ],
+    push_cast,
+    ring_nf },
+end
+
+
+/-- The sequence `c n` has limit `1/2` -/
+lemma rest_has_limit_one_half : tendsto (λ (n : ℕ), (n : ℝ) / (2 * n + 1)) at_top (𝓝 (1 / 2)) :=
```

Comment:
This lemma name is not very good (what is `rest`? Also the statement does not mention `has_limit`, but `tendsto`, so this is what should appear in the lemma name). Something like `tendsto_self_div_two_mul_self_add_one` is quite verbose, but since it will not be used elsewhere I think it should be fine.